pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -668,12 +673,16 @@ func (ctrl *Controller) syncImageConfig(key string) error {
 			if err != nil {
 				return err
 			}
+			rawRegistriesIgn, err := ctrlcommon.EncodeIgnitionConfigSpecV2(registriesIgn)
+			if err != nil {
+				return fmt.Errorf("could not encode registries Ignition config: %v", err)
+			}
 			mc, err := ctrl.client.MachineconfigurationV1().MachineConfigs().Get(managedKey, metav1.GetOptions{})
 			if err != nil && !errors.IsNotFound(err) {
 				return fmt.Errorf("could not find MachineConfig: %v", err)
 			}
 			isNotFound := errors.IsNotFound(err)
-			if !isNotFound && equality.Semantic.DeepEqual(registriesIgn, mc.Spec.Config) {
+			if !isNotFound && equality.Semantic.DeepEqual(rawRegistriesIgn, mc.Spec.Config) {
```

Comment:
Are references to `mc.Spec.Config` the exact same as `mc.Spec.Config.Raw`? I'm a little confused since now we use `.Raw` for the object, but here we are comparing directly.

Reply:
this should indeed be comparing to `mc.Spec.Config.Raw`. I guess isn't caught because the RawExtension is something of an artificial type that doesn't add any extra bytes (i.e. wire-format stays exactly the same).